logic/strategy/card_utils.go

```diff
@@ -36,41 +38,41 @@ func otherOptions(desired int, avoid map[model.Card]struct{}) [][]model.Card {
 	return options
 }
 
-func chooseFrom(desired int, hand []model.Card) [][]model.Card { //nolint:gocyclo
-	if desired > len(hand) || desired > 4 || desired <= 0 {
-		return nil
+func chooseFrom(k int, hand []model.Card) ([][]model.Card, error) {
+	if len(hand) > 6 || k < 1 || k > len(hand) {
+		return nil, errors.New(`invalid input`)
```

Comment:
(same comment here)
on this one, maybe split into two if checks:
`hand too large` vs. `developer error: invalid k`